### DISCUSSION TITLE
Microfrontend integration (Single-spa or Qiankun) with Nuxt 3

### DISCUSSION THREAD

**Discussion by HundredBeans** [nuxt/nuxt#18622] [Questions]
I wanted to create a micro-frontend app with Nuxt 3. I will be using [Qiankun](https://qiankun.umijs.org/) as the micro-frontend framework. 

The problem is, Qiankun requires the sub-app to export certain lifecycle functions from the entry file https://qiankun.umijs.org/guide/getting-started#sub-application. I found a [Nuxt module](https://github.com/FEMessage/nuxt-micro-frontend) that could handle this, but it only works with Nuxt 2. How these module works is by changing the default template for the client.js and adding the required lifecycle functions to that template. After reviewing the Nuxt 3 source code, I think this method is no longer possible due to changes in the implementation.

My question is, how do I implement that in Nuxt 3? Is it possible to modify the entry file? Or maybe there is a better approach for this case?

*For additional reference and information, there is also a similar post that I found for the Nuxt 2 implementation here: https://github.com/nuxt/nuxt/issues/8970

I'd greatly appreciate any help on this matter. Thank you!

**Comment by SMNaik123** [comment]
Any good microfrontend frameworks available for Nuxt3 ?

**Comment by hpy-lhj** [comment]
Excuse me, have you solved this problem? I have encountered the same problem.

**Comment by luis-abeno** [comment]
Any news?

I'm also looking for a solution integrating Qiankun with Nuxt 3.
